### ISSUE TITLE
[lsp] The LSP (XTEXT) server announces too many capabilities 

### ISSUE THREAD

**Issue by MHerbst** [eclipse/smarthome#6654]
If the LSP is activated in VSCode the context menus for the config files (.items/.sitemap/ ...) contain some options that are not working (Go to Definition, Peek Definition. Find/List All References, Format Document, ...) because of missing implementations. Some of them do nothing (e.g. "Format") some of them produce error messages (Go To Definition).

They are all active because in the list of server capabilities (https://github.com/eclipse/xtext-core/blob/f21f2dcf6305b6501b5d74d6f528a57d2bbb5c27/org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend#L148) all these capabilities are set to "true" by default.

It would be better if only the implemented functions were enabled.

**Comment by MHerbst** [comment]
In the VSCode repo we are discussing the implementation of something like an LSP proxy to enhance performance for some functions (https://github.com/openhab/openhab-vscode/issues/95#issuecomment-445305369). If this works, the supported capabilities could be configured easily by the proxy.

**Comment by maggu2810** [comment]
I don't use LSP myself, but how is this related to a LSP proxy at all?
If the server can provide function and the provided function is published by capabilities, I would expect that only the capabilities are set, that are supported.
If a proxy adds additional functions it can perhaps modify the capability response and add that functions.
WDYT?

**Comment by MHerbst** [comment]
I agree with you. In the next days, I will have a closer look into the LSP implementation and try to figure out how this exactly works.
